src/framework/emqttb_worker.erl

```diff
@@ -444,6 +453,16 @@ connect_fun()->
             connect
     end.
 
+-spec broker_host(emqttb:host_selection(), integer()) -> {string(), inet:port_number()}.
+broker_host(random, _HostShift) ->
+  Hosts = broker_hosts(),
+  lists:nth(rand:uniform(length(Hosts)), Hosts);
```

Comment:
Please use shuffle instead of `nth`, so the entire list of hosts is passed to emqtt

Reply:
P.S. Maybe this can be done in emqtt repo?